architecture/builds.adoc

```diff
@@ -44,3 +44,52 @@ STI builds are a replacement for the OpenShift v2-like developer experience. The
 == Custom Build
 Custom builds are the most sophisticated version of builds, allowing developers to define a builder image which is responsible for the entire process of the build. The custom builder image is a plain Docker image within which the author embeds the logic of the desired build process, such as building RPMs or building
 base Docker images.
+
+[#using-docker-credentials-for-pushing-images]
+== Using Docker credentials for pushing images
+
+In case you want to push the output image into private Docker Registry that
+require authentication or Docker Hub, you have to supply the `dockercfg` file
+with valid Docker Registry credentials.
+
+The `.dockercfg` JSON file usually exists in your home directory and it has following
+format:
+
+```
+{"https://index.docker.io/v1/":{"auth":"encrypted_password","email":"foo@bar.com"}}
+```
+
+The 'https://index.docker.io/v1' is the URL of the registry. You can define
+multiple Docker registries entries in this file.
+
+Kubernetes supports
+https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/secrets.md[Secret]
+resource, you can use to store your passwords and configuration.
+In order to make Build use your `.dockercfg` file for pushing the output image,
+you have to create the Secret first. The Secret name should be 'dockercfg' and
```

Comment:
the secret name doesn't have to be dockercfg, right?  it can be anything, as long as you then put the same thing in PushSecretName